pydoctor/astbuilder.py

```diff
@@ -47,28 +45,142 @@ def node2fullname(expr: Optional[ast.expr], ctx: model.Documentable) -> Optional
+    """Binds the arguments of a function call to that function's signature.
+    @raise TypeError: If the arguments do not match the signature.
+    """
+    kwargs = {
+        kw.arg: kw.value
+        for kw in call.keywords
+        if kw.arg is not None
+        }
+    return sig.bind(*call.args, **kwargs)
+
+
+_attrs_signature = signature(attrs)
+
+def _uses_auto_attribs(call: ast.Call, module: model.Module) -> bool:
+    """Does the given L{attr.s()} decoration contain C{auto_attribs=True}?
+    @param call: AST of the call to L{attr.s()}.
+        This function will assume that L{attr.s()} is called without
+        verifying that.
+    @param module: Module that contains the call, used for error reporting.
+    @return: L{True} if L{True} is passed for C{auto_attribs},
+        L{False} in all other cases: if C{auto_attribs} is not passed,
+        if an explicit L{False} is passed or if an error was reported.
+    """
+    try:
+        args = bind_args(call, _attrs_signature)
+    except TypeError as ex:
+        module.report(
+            f"Invalid arguments for attr.s(): {ex}",
+            lineno_offset=call.lineno
+            )
+        return False
     else:
-        return None
+        auto_attribs_expr = args.arguments.get('auto_attribs')
+        if auto_attribs_expr is None:
+            return False
+        try:
+            value = ast.literal_eval(auto_attribs_expr)
+        except ValueError:
+            module.report(
+                'Unable to figure out value for "auto_attribs" argument '
+                'to attr.s(), maybe too complex',
+                lineno_offset=call.lineno
+                )
+            return False
+        if isinstance(value, bool):
+            return value
+        else:
+            module.report(
+                f'Value for "auto_attribs" argument to attr.s() '
+                f'has type "{type(value).__name__}", expected "bool"',
+                lineno_offset=call.lineno
+                )
+            return False
+
+
+def is_attrib(expr: Optional[ast.expr], ctx: model.Documentable) -> bool:
+    """Does this expression return an C{attr.ib}?"""
+    return isinstance(expr, ast.Call) and node2fullname(expr.func, ctx) in (
+        'attr.ib', 'attr.attrib', 'attr.attr'
+        )
+
+
+_attrib_signature = signature(attrib)
+
+def attrib_args(expr: ast.expr, ctx: model.Documentable) -> Optional[BoundArguments]:
+    """Get the arguments passed to an C{attr.ib} definition.
+    @return: The arguments, or L{None} if C{expr} does not look like
+        an C{attr.ib} definition or the arguments passed to it are invalid.
+    """
+    if isinstance(expr, ast.Call) and node2fullname(expr.func, ctx) in (
+            'attr.ib', 'attr.attrib', 'attr.attr'
+            ):
+        try:
+            return bind_args(expr, _attrib_signature)
+        except TypeError as ex:
+            ctx.module.report(
+                f"Invalid arguments for attr.ib(): {ex}",
```

Comment:
Yes, it can be `attr.ib`, `attr.attrib` or `attr.attr`, although the last one is deprecated.

I don't think "alias" is very clear either. An alternative would be to name them explicitly: `Invalid arguments for attr.ib/attr.attrib():`, but that doesn't look great either in my opinion.

Another option would be to insert the actual name that the code uses, but then we'd have to quote the name to be consistent: `Invalid arguments for "attr.ib":` or `Invalid arguments for "attr.attrib":`. And the variation would make the message slightly harder to find in search engines. Also the messages for `attr.s` would need a similar treatment.

To me none of the options stands out as clearly superior, so I'd be fine with leaving it as it is. But if you think that showing the name as used in the code is much better, I'm willing to change it.

Reply:
No need to put extra work on this.